components/authz-service/storage/postgres/migration/sql/55_add_db_id_statements.up.sql

```diff
@@ -0,0 +1,180 @@
+BEGIN;
+ALTER TABLE iam_policy_statements
+    DROP CONSTRAINT iam_policy_statements_pkey,
+    DROP CONSTRAINT iam_policy_statements_policy_id_fkey;
+ALTER TABLE iam_statement_projects
+    DROP CONSTRAINT iam_statement_projects_statement_id_fkey;
+ALTER TABLE iam_policy_statements
+    ADD COLUMN policy_db_id SERIAL;
+UPDATE
+    iam_policy_statements ips
+SET
+    policy_db_id = (
+        SELECT
+            db_id
+        FROM
+            iam_policies
+        WHERE
+            id = ips.policy_id);
+ALTER TABLE iam_statements
+    DROP CONSTRAINT iam_statements_pkey;
+ALTER TABLE iam_statements
+    ADD COLUMN db_id SERIAL PRIMARY KEY UNIQUE,
+    ADD COLUMN policy_id INTEGER REFERENCES iam_policies(db_id) ON DELETE CASCADE DEFERRABLE,
+    ADD UNIQUE (id);
+UPDATE
+    iam_statements iams
+SET
+    policy_id = (
+        SELECT
+            policy_db_id
+        FROM
+            iam_policy_statements
+        WHERE
+            statement_id = iams.id);
+DROP TABLE iam_policy_statements CASCADE;
+ALTER TABLE iam_statement_projects
+    ADD UNIQUE (statement_id, project_id),
+    ADD CONSTRAINT iam_statement_projects_statement_id_fkey FOREIGN KEY (statement_id) REFERENCES iam_statements(id) ON DELETE CASCADE DEFERRABLE;
+
+-- add helper for statement db_id lookup
+CREATE FUNCTION statement_db_id (_id UUID)
+    RETURNS INTEGER
+    AS $$
+    SELECT
+        db_id
+    FROM
+        iam_statements
+    WHERE
+        id = _id;
+$$
+LANGUAGE SQL;
+
+ CREATE OR REPLACE FUNCTION query_policy (_policy_id TEXT, _projects_filter TEXT[])
+    RETURNS json
```

Comment:
I wanted to go through the exercise of seeing what is involved in converting this style of query we had been using as a team into the newer style that @srenatus started in his recent PR (https://github.com/chef/automate/pull/878), and also figured I could save you some grunt work, @blakestier. So I tackled `query_policy` and I believe I've got it working. Can chat about it when you have time.

Reply:
I'd propose another round for this. Let's merge this, if we find it OK, instead of letting scope creep...

Reply:
@msorens That's _awesome_. Once we merge this I'll give you a heads up and we can get your change in before doing any more db work! 😅